model/textparse/openmetricsparse.go

```diff
@@ -222,9 +223,99 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 // CreatedTimestamp returns nil as it's not implemented yet.
 // TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	var lbs labels.Labels
+	p.Metric(&lbs)
+	lbs = lbs.DropMetricName()
+
+	newParser := deepCopyParser(p)
+loop:
+	for {
+		switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			// continue instead of return nil until we get new series/labels. can happen for histograms, summaries
+			// Check _created suffix
+			var newLbs labels.Labels
+			newParser.Metric(&newLbs)
+			name := newLbs.Get(model.MetricNameLabel)
+			if !strings.HasSuffix(name, "_created") {
+				continue
+			}
+
+			// TODO: potentially broken? Missing type?
+			if newParser.mName != p.mName {
+				return nil
+			}
+
+			// edge case: if gauge_created of unknown type -> skip parsing
+			newLbs = newLbs.DropMetricName()
+			if !labels.Equal(lbs, newLbs) {
+				return nil
+			}
+
+			// TODO: for histograms
+			// if t, _ := newParser.Next(); t != EntrySeries {
+			// 	return nil
+			// }
+
+			// gauge_created is a metric
+			ct := int64(newParser.val)
+			_, err := p.Next()
+			if err != nil {
+				return nil
+			}
```

Comment:
Could you explain your thoughts when calling p.Next() here? Is it so the original parser skips the `_created` line and does not parse it as a metric?

If that's the case, it won't work for histograms/summaries since they often have multiple lines until they reach `_created`

Reply:
what if we add a new boolean to the parser

```diff
type OpenMetricsParser struct {
	l       *openMetricsLexer
	builder labels.ScratchBuilder
	series  []byte
	text    []byte
	mtype   model.MetricType
	val     float64
	ts      int64
	hasTS   bool
	start   int
	// offsets is a list of offsets into series that describe the positions
	// of the metric name and label names and values for this series.
	// p.offsets[0] is the start character of the metric name.
	// p.offsets[1] is the end of the metric name.
	// Subsequently, p.offsets is a pair of pair of offsets for the positions
	// of the label name and value start and end characters.
	offsets []int

	eOffsets      []int
	exemplar      []byte
	exemplarVal   float64
	exemplarTs    int64
	hasExemplarTs bool
+	returnCreatedLine bool
}
```

That is always false for original parser, but true when you're deep copying the new parser in `CreatedTimestamp()`. During p.Next(), we skip lines ending with `_created` suffix if `returnCreatedLine` is false

Reply:
covered via our pair programming session